xdsl/dialects/csl_/csl_stencil.py

```diff
@@ -0,0 +1,270 @@
+from collections.abc import Sequence
+from itertools import pairwise
+from typing import Annotated, cast
+
+from xdsl.dialects import builtin, memref, stencil
+from xdsl.dialects.builtin import IntegerAttr, IntegerType, TensorType
+from xdsl.dialects.experimental import dmp
+from xdsl.ir import Attribute, Dialect, Operation, ParametrizedAttribute, SSAValue
+from xdsl.irdl import (
+    ConstraintVar,
+    IRDLOperation,
+    Operand,
+    ParameterDef,
+    attr_def,
+    irdl_attr_definition,
+    irdl_op_definition,
+    operand_def,
+    opt_attr_def,
+    opt_prop_def,
+    prop_def,
+    result_def,
+)
+from xdsl.parser import AttrParser, Parser
+from xdsl.printer import Printer
+from xdsl.traits import HasAncestor, Pure
+from xdsl.utils.exceptions import VerifyException
+from xdsl.utils.hints import isa
+
+
+@irdl_attr_definition
+class ExchangeDeclarationAttr(ParametrizedAttribute):
+    """
+    A simplified version of dmp.exchange, from which it should be lowered
+    """
+
+    name = "csl_stencil.exchange"
+
+    neighbor_: ParameterDef[builtin.DenseArrayBase]
```

Comment:
Is there a way to avoid the tralining underscore? It just looks a little weird and isn't super easy to distinguish from its tuple counterpart. I'd probably recommend removing the underscore here, and renaming `neighbor` to `neighbor_tuple`, but an alternative would be to make this something like `neighbor_attr`, or something along those lines.

Reply:
renamed to remove trailing underscore